### PR TITLE
Fixes 3353: Snapshot  inconsistent for API/UI

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary

Snapshot create-on-edit is inconsistent between API and UI 
This was caused by the URL not being checked correctly before enqueueing snapshot/introspect tasks. 
This card corrects that issue.

## Testing steps

- From the api or UI, update a repository (patch) with the original URL in the request.  
Without this change, introspection/snapshot tasks will always be enqueued. 
With it, the tasks will correctly, be ignored, as those two tasks are contingent on a URL change.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-3353

**Comment by swadeley** [comment]
lgtm

File: pkg/handler/repositories.go [hunk 1]
```diff
@@ -313,7 +313,9 @@ func (rh *RepositoryHandler) update(c echo.Context, fillDefaults bool) error {
 		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error updating repository", err.Error())
 	}
 
-	if repoParams.URL != nil && repoConfig.URL != *repoParams.URL {
+	urlIsDifferent := repoConfig.URL != *repoParams.URL && urlUpdated
```
Comment (rverdile) [inline pkg/handler/repositories.go:316]: What do you think moving this comparison logic into the dao Update method and using the `urlUpdated` variable to represent if the URL has changed. I believe that was the intention of that return value.

Comment (Andrewgdewar) [reply pkg/handler/repositories.go:316]: Hmm good point. I'll do that.

Comment (Andrewgdewar) [reply pkg/handler/repositories.go]: Thanks @rverdile, much cleaner.

File: pkg/handler/repositories.go [hunk 2]
```diff
@@ -313,7 +313,7 @@ func (rh *RepositoryHandler) update(c echo.Context, fillDefaults bool) error {
 		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error updating repository", err.Error())
 	}
 
-	if repoParams.URL != nil && repoConfig.URL != *repoParams.URL {
+	if urlUpdated && repoParams.URL != nil && repoConfig.URL != *repoParams.URL {
```
Comment (rverdile) [inline pkg/handler/repositories.go:316]: it seems like `urlUpdated` already checks `repoParams.URL != nil && repoConfig.URL != *repoParams.URL`, so I think you can remove everything to the right of `urlUpdated`.

Comment (Andrewgdewar) [reply pkg/handler/repositories.go:316]: You are correct good sir! 
Thank you.

**Review by rverdile — APPROVED** [review]
lgtm!

**Pull request merged**